### PR TITLE
Add ability to ingest GitHub repository information (Issue #228)

### PR DESCRIPTION
**Pull request by tayasteere**
Initial commit to allow GitHub repo information
Includes tests and documentation

### CODE REVIEW

File: cartography/intel/github/__init__.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+from requests import exceptions
+
+from cartography.intel.github.github import sync_github
+from cartography.util import run_cleanup_job
+
+logger = logging.getLogger(__name__)
+
+# API key for GitHub
+GITHUB_AUTH_DATA = os.environ.get('CREDENTIALS_GITHUB_AUTH_DATA')
```
[the remark this answers is not in the document]
Comment (achantavy) [reply cartography/intel/github/__init__.py]: We would also need to add instructions to the Readme how a user can setup a Github sync .

File: cartography/intel/github/github.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+            WITH user
+            MATCH (repo:GitHubRepository{id: {RepoId}})
+            MERGE (user)-[r:OWNER]->(repo)
+            ON CREATE SET r.firstseen = timestamp()
+            SET r.lastupdated = {UpdateTag}"""
+
+        account_type = {'User': "GitHubUser", 'Organization': "GitHubOrganization"}
+
+        session.run(
+            ingest_owner.replace(
```
Comment (achantavy) [inline cartography/intel/github/github.py]: Replace this method of string replacement with string Templating instead.  See https://github.com/lyft/cartography/blob/d595b845c5f9c28f88292e252a0803fa63f1553b/cartography/intel/gcp/crm.py#L179 for an example

File: docs/schema/github.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+| id | The GitHub organization id. These are not unique across GitHub instances, so are prepended with the API URL the id applies to |
+| username | Name of the organization |
+
+
+### Relationships
+
+- GitHubOrganizations own GitHubRepositories.
+
+    ```
+    (GitHubOrganization)-[OWNER]->(GitHubRepository)
```
Comment (achantavy) [inline docs/schema/github.md]: @SecPrez do you think the relationship here should point _toward the owner_ instead?

File: docs/schema/github.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+| id | The GitHub user id. These are not unique across GitHub instances, so are prepended with the API URL the id applies to |
+| username | Name of the user |
+
+
+### Relationships
+
+- GitHubUsers own GitHubRepositories.
+
+    ```
+    (GitHubUser)-[OWNER]->(GitHubRepository)
```
Comment (achantavy) [inline docs/schema/github.md]: @SecPrez Ditto

File: cartography/intel/github/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tayasteere) [reply cartography/intel/github/__init__.py]: done

File: docs/schema/github.md (hunk 3 above)
Comment (tayasteere) [reply docs/schema/github.md]: @achantavy @SecPrez this is the schema currently used internally, please confirm if you want it changed for open source

File: cartography/intel/github/github.py (hunk 2 above)
Comment (tayasteere) [reply cartography/intel/github/github.py]: done

File: cartography/intel/crxcavator/crxcavator.py [hunk 5]
```diff
@@ -219,6 +219,9 @@ def transform_user_extensions(user_extension_json):
                 'name': details[1]['name'],
             })
             for user in details[1]['users']:
+                if user is None:
+                    print(f"bad user for {extension_id}|{version}")
```
[the remark this answers is not in the document]
Comment (tayasteere) [reply cartography/intel/crxcavator/crxcavator.py]: pulling from this PR and will be part of crxcavator PR. Was artifact of prior local branch I think

File: docs/schema/github.md (hunk 3 above)
Comment (achantavy) [reply docs/schema/github.md]: Ugh, if that's how we already have it then sure, I won't block on this for now.

Comment (tayasteere) [reply docs/schema/github.md]: changed, will address breaks downstream if necessary

**Pull request merged**